### PR TITLE
#160577586 Enable users to filter articles

### PR DESCRIPTION
**Pull request by bevkololi**
**What does this PR do?**

It enables users to filter articles using specific parameters.

**Description of Task to be completed?**

Users may want to get desired articles quickly. This PR allows them to filter articles using parameters i.e tags and authors' usernames.

**How should this be manually tested?**

- Open this PR's review app
- Click on the search icon in the Navbar
- Click on filter icon
- Fill details, i.e tag or author or both
- The articles should appear as desired i.e (you will get articles by your desired author or tags)

**Any background context you want to provide?**

This feature is heavily integrated with the search feature. By this I mean that one can filter using a parameter and then search anything on the results that will be provided.

**What are the relevant pivotal tracker stories?**

https://www.pivotaltracker.com/story/show/160577586

**Screenshots (if appropriate)**

This is what you see when you navigate to the search page
screen shot 2018-11-29 at 11 29 29

Input the filter parameters. There is also autocomplete as a user types
screen shot 2018-11-29 at 11 29 55

Click apply and the articles should appear on your right
screen shot 2018-11-29 at 11 30 09

**Questions:**

N/A

### CODE REVIEW

File: src/containers/FilterDropDown/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    fetchTags();
+    fetchAuthors();
+  }
+
+  selectAll = (e) => {
+    e.target.select();
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    this.authorsInstance = Materialize.Autocomplete.init(this.authors, { data: nextProps.authors });
```
Comment (paulrimiru) [inline src/containers/FilterDropDown/index.js:22]: Add tests for this functionality to actually check that the change in props reflects

File: src/pages/Search/search.test.js [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
 const props = {
   fetchArticles: jest.fn(),
+  onSearch: jest.fn(),
+  onFilterChanged: jest.fn(),
+  filtersChanged: jest.fn(),
 };
 
-describe('the search container', () => {
+
+describe('<Search />) container', () => {
```
Comment (paulrimiru) [inline src/pages/Search/search.test.js]: The test cases should read as a sentence, please change this to ` The search container`

File: src/pages/Search/search.test.js [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
   const wrapper = mount(<Provider store={store}><Search {...props} /></Provider>);
 
   beforeEach(() => {
     wrapper.setProps({ ...props });
   });
 
-  it('renders without crashing', () => {
+  it('renders <Search /> without crashing', () => {
```
Comment (paulrimiru) [inline src/pages/Search/search.test.js]: remove `<Search />` from the test case title

File: src/pages/Search/index.js [hunk 4]
```diff
@@ -5,49 +5,116 @@ import { connect } from 'react-redux';
...
+        "search");
     }
 
+    renderFilters = (author, tag) => (
+      <div className="card">
+        <div className="card-content">
+          <h4>Filters</h4>
+          {
+            author && author.username && (
+              <>
```
Comment (paulrimiru) [inline src/pages/Search/index.js]: Use `Fragment` instead, for readability

**Review by paulrimiru — CHANGES_REQUESTED** [review]

File: src/containers/FilterDropDown/index.js (hunk 1 above)
Comment (bevkololi) [reply src/containers/FilterDropDown/index.js:22]: Researching on this

File: src/pages/Search/search.test.js (hunk 2 above)
Comment (bevkololi) [reply src/pages/Search/search.test.js]: Done

File: src/pages/Search/search.test.js (hunk 3 above)
Comment (bevkololi) [reply src/pages/Search/search.test.js]: Done

File: src/pages/Search/index.js (hunk 4 above)
Comment (bevkololi) [reply src/pages/Search/index.js]: Done

File: src/containers/FilterDropDown/index.js (hunk 1 above)
Comment (bevkololi) [reply src/containers/FilterDropDown/index.js:22]: This is also done @PaulKariukiRimiru

**Review by paulrimiru — APPROVED** [review]

**Pull request merged**